usr/src/lib/libzpool/common/kernel.c

```diff
@@ -346,7 +346,10 @@ cv_timedwait_hires(kcondvar_t *cv, kmutex_t *mp, hrtime_t tim, hrtime_t res,
 	ASSERT(flag == 0);
 
 top:
-	delta = tim - gethrtime();
+	delta = tim;
+	if (flag & CALLOUT_FLAG_ABSOLUTE)
```

Comment:
As part of this fix I was trying to make cv_timedwait_hires() more like the kernel version and that's why I added CALLOUT_FLAG_ABSOLUTE. The primary fix, however, is that userland consumers of cv_timedwait_hires() were always getting absolute time semantics even though all of them wanted relative time semantics. 

For my testing I had left the ASSERT in to determine if we had any zfs consumers that were relying on absolute time semantics. It turns out that we don't have any userland consumers that use CALLOUT_FLAG_ABSOLUTE, so I should have removed the ASSERT so that new consumers could use this functionality. Thanks for catching this.